selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/miscellaneous/FileStructureNodesTest.java

```diff
@@ -174,46 +176,50 @@ void checkFileStructureFilter() {
 
   @Test(priority = 1)
   public void checkFileStructureNodes() {
-    // check work nodes in the 'file structure' by double click
+    // check work nodes in the File Structure form by double click
     fileStructure.waitExpectedTextInFileStructure(ITEMS_CLASS);
     fileStructure.waitExpectedTextInFileStructure(ITEMS_INNER_CLASS);
-    fileStructure.selectItemInFileStructureByDoubleClick(INNER_CLASS_NAME);
+
+    // open and close nodes in the File Structure form by click
+    fileStructure.clickOnIconNodeInFileStructure(INNER_CLASS_NAME);
     fileStructure.waitExpectedTextIsNotPresentInFileStructure(ITEMS_INNER_CLASS);
+    fileStructure.clickOnIconNodeInFileStructure(INTERFACE_NAME);
+    fileStructure.waitExpectedTextIsNotPresentInFileStructure(ITEMS_INTERFACE);
+
+    fileStructure.clickOnIconNodeInFileStructure(INTERFACE_NAME);
     fileStructure.waitExpectedTextInFileStructure(ITEMS_INTERFACE);
+    fileStructure.clickOnIconNodeInFileStructure(INNER_CLASS_NAME);
+    fileStructure.waitExpectedTextInFileStructure(ITEMS_INNER_CLASS);
+    fileStructure.waitExpectedTextInFileStructure(ITEMS_CLASS);
+
+    // open and close nodes in the File Structure form by double click
+    fileStructure.selectItemInFileStructureByDoubleClick(INNER_CLASS_NAME);
+    fileStructure.waitExpectedTextIsNotPresentInFileStructure(ITEMS_INNER_CLASS);
+    fileStructure.selectItemInFileStructureByDoubleClick(INNER_CLASS_NAME);
+    fileStructure.waitExpectedTextInFileStructure(ITEMS_INNER_CLASS);
+    fileStructure.waitExpectedTextInFileStructure(ITEMS_CLASS);
```

Comment:
It's really hard to understand where node is closing. From this point it would be better to comment the code separately.